tests/scripts/thread-cert/Cert_6_2_02_NewPartition.py

```diff
@@ -88,10 +94,37 @@ def test(self):
         self.assertEqual(self.nodes[ROUTER1].get_state(), 'router')
         self.assertEqual(self.nodes[ED].get_state(), 'child')
 
+        self.collect_ipaddrs()
         addrs = self.nodes[ED].get_addrs()
         for addr in addrs:
             self.assertTrue(self.nodes[ROUTER1].ping(addr))
 
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ED = pv.vars['ED']
+        _router1_pkts = pkts.filter_wpan_src64(ROUTER_1)
+        _ed_pkts = pkts.filter_wpan_src64(ED)
+
+        # Step 1: Ensure that the DUT successfully attached to Router_1
+        _ed_pkts.filter_mle_cmd(consts.MLE_CHILD_ID_REQUEST).filter_wpan_dst64(
+            ROUTER_1).must_next()
+
+        # Step 2-6 : remove Leader and make ROUTER_2 as new Leader
+        # Step 7: The DUT MUST send a MLE Child Update Request to Router_1
+        _ed_pkts.filter_mle_cmd(consts.MLE_ADVERTISEMENT).must_not_next()
+        _ed_pkts.filter_mle_cmd(consts.MLE_CHILD_ID_REQUEST).must_not_next()
+        _ed_pkts.filter_wpan_dst64(ROUTER_1).filter_mle_cmd(13).must_next(
+        ).must_verify(lambda p: {1, 0, 11} < set(p.mle.tlv.type))
+
+        # Step 8: The DUT MUST respond with ICMPv6 Echo Reply
+        ed_mleid = pv.vars['ED_MLEID']
+        router1_mleid = pv.vars['ROUTER_1_MLEID']
+        _ed_pkts.filter(lambda p: p.ipv6.dst == router1_mleid and p.ipv6.src ==
+                        ed_mleid).filter_ping_reply().must_next()
```

Comment:
nit: the code can be further simplified to:
```suggestion
        _ed_pkts.filter(lambda p: p.ipv6.dst == {ROUTER_1_MLEID} and p.ipv6.src ==
                        {ED_MLEID}, **pv.vars).filter_ping_reply().must_next()
```